packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -472,259 +468,258 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
+        // there may still be markers due to inlined dynamic imports, remove
+        // all the markers regardless
+        let markerStartPos = indexOfRegexp(code, preloadMarkerRE)
+        while (markerStartPos >= 0) {
+          if (!rewroteMarkerStartPos.has(markerStartPos)) {
+            s.update(
+              markerStartPos,
               markerStartPos + preloadMarker.length,
+              'void 0',
             )
           }
+          markerStartPos = indexOfRegexp(
+            code,
+            preloadMarkerRE,
+            markerStartPos + preloadMarker.length,
+          )
+        }
 
-          if (s.hasChanged()) {
-            chunk.code = s.toString()
-            if (buildSourcemap && chunk.map) {
-              const nextMap = s.generateMap({
-                source: chunk.fileName,
-                hires: 'boundary',
-              })
-              const map = combineSourcemaps(chunk.fileName, [
-                nextMap as RawSourceMap,
-                chunk.map as RawSourceMap,
-              ]) as SourceMap
-              map.toUrl = () => genSourceMapUrl(map)
-
-              const originalDebugId = chunk.map.debugId
-              chunk.map = map
-
-              if (buildSourcemap === 'inline') {
-                chunk.code = chunk.code.replace(
-                  convertSourceMap.mapFileCommentRegex,
-                  '',
-                )
-                chunk.code += `\n//# sourceMappingURL=${genSourceMapUrl(map)}`
-              } else {
-                if (originalDebugId) {
-                  map.debugId = originalDebugId
-                }
-                const mapAsset = bundle[chunk.fileName + '.map']
-                if (mapAsset && mapAsset.type === 'asset') {
-                  mapAsset.source = map.toString()
-                }
-              }
-            }
+        if (!s.hasChanged()) {
+          continue
+        }
+
+        chunk.code = s.toString()
+
+        if (!buildSourcemap || !chunk.map) {
+          continue
+        }
+
+        const nextMap = s.generateMap({
+          source: chunk.fileName,
+          hires: 'boundary',
+        })
+        const map = combineSourcemaps(chunk.fileName, [
+          nextMap as RawSourceMap,
+          chunk.map as RawSourceMap,
+        ]) as SourceMap
+        map.toUrl = () => genSourceMapUrl(map)
+
+        const originalDebugId = chunk.map.debugId
+        chunk.map = map
+
+        if (buildSourcemap === 'inline') {
+          chunk.code = chunk.code.replace(
+            convertSourceMap.mapFileCommentRegex,
+            '',
+          )
+          chunk.code += `\n//# sourceMappingURL=${genSourceMapUrl(map)}`
+        } else {
+          if (originalDebugId) {
+            map.debugId = originalDebugId
+          }
+          const mapAsset = bundle[chunk.fileName + '.map']
+          if (mapAsset && mapAsset.type === 'asset') {
+            mapAsset.source = map.toString()
           }
         }
       }
```

Comment:
I removed nesting logic where possible.

Recommended to review with white-space hidden.

Reply:
The handlers was moved to `.handler` intentionally by https://github.com/vitejs/vite/pull/19586 to reduce the diff with rolldown powered Vite. Would you revert that part?

Reply:
Done and ready for review!